### PR TITLE
#210 added support for server-side rendering

### PR DESCRIPTION
**Pull request by temaptz**
Browser-specific api (WINDOW, DOCUMENT) used with special wrappers.
``@Inject(DOCUMENT) private _document: Document`` instead of document

``isPlatformBrowser(this._platformId)`` for checking before adding event listeners for window

### CODE REVIEW

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts [hunk 1]
```diff
@@ -320,4 +323,11 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
     this._renderer.setStyle(this.nsmContent.nativeElement, 'top', offsetTop + 'px');
     this._renderer.setStyle(this.nsmContent.nativeElement, 'left', offsetLeft + 'px');
   }
+
+  /**
+   * Is current platform browser
+   */
+  public isBrowser(): boolean {
```
Comment (khylias) [inline src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: This method is only used in the component, so you can define it as private and make it a getter.

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts [hunk 2]
```diff
@@ -183,7 +186,7 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
         me.onAnyCloseEventFinished.emit(me);
       }, this.hideDelay);
 
-      window.removeEventListener('keyup', this.escapeKeyboardEvent);
+      this.isBrowser() && window.removeEventListener('keyup', this.escapeKeyboardEvent);
```
Comment (khylias) [inline src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: For all your browser tests, it is preferable to detail statements to improve readability as : 
```
if (this.isBrowser ) {
 // do some stuff
}
```

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts (hunk 1 above)
Comment (temaptz) [reply src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: Done.

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts (hunk 2 above)
Comment (temaptz) [reply src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: Done for all browser tests.

File: src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts [hunk 3]
```diff
@@ -72,7 +73,9 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
   constructor(
     private _renderer: Renderer2,
     private _changeDetectorRef: ChangeDetectorRef,
-    private _ngxSmartModalService: NgxSmartModalService
+    private _ngxSmartModalService: NgxSmartModalService,
+    @Inject(DOCUMENT) private _document: Document,
+    @Inject(PLATFORM_ID) private _platformId: Object
```
Comment (khylias) [inline src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: TravisCI signals this `Object` must be `object`.

Comment (temaptz) [reply src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts]: fixed

**Comment by maximelafarie** [comment]
@khylias do you think is it rightful to add unit tests for the `isBrowser` getter or not?

**Comment by khylias** [comment]
@maximelafarie  It simply returns an angular/core injection, I don't think we need to test it.

**Comment by maximelafarie** [comment]
Since we've merged https://github.com/biig-io/ngx-smart-modal/pull/215 into master, your PR has conflicts with the new code. I let you understand the new way services are handled by modal component and do the appropriate changes. I apologize for the inconvenience and I'm at your disposal for further help if needed.
Regards! 🖖

**Comment by temaptz** [comment]
@maximelafarie , conflicts are resolved

**Comment by maximelafarie** [comment]
Closing it in favor of #243. Many thank for your contribution! Cheers! 🖖

**Pull request closed without merging**